### PR TITLE
Google calendar

### PR DESCRIPTION
**Pull request by Tarraann**
Google Calendar Tool

### CODE REVIEW

File: superagi/tools/google_calendar/event_details_calendar.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if service["success"]:
+            service = service["service"]
+        else:
+            return f"Kindly connect to Google Calendar"
+        if event_id == "None":
+            return f"Add Event ID to fetch details of an event from Google Calendar"
+        else:
+            decoded_id = base64.b64decode(event_id)
+            eid = decoded_id.decode("utf-8")
+            eid  = eid.split(" ", 1)[0]
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/event_details_calendar.py:27]: can be done in single line.
base64.b64decode(event_id).decode("utf-8").split(" ", 1)[0]

File: superagi/tools/google_calendar/list_calendar_events.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        file_name = "Google_Calendar_" + file + ".csv"
+        final_path = file_name
+        root_dir = get_config('RESOURCES_OUTPUT_ROOT_DIR')
+        if root_dir is not None:
+            root_dir = root_dir if root_dir.startswith("/") else os.getcwd() + "/" + root_dir
+            root_dir = root_dir if root_dir.endswith("/") else root_dir + "/"
+            final_path = root_dir + file_name
+        else:
+            final_path = os.getcwd() + "/" + file_name
+        with open(final_path, "w") as file:
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/list_calendar_events.py:79]: use ResourceManager class. Add write_csv_file method in that for this use case.

Comment (Tarraann) [reply superagi/tools/google_calendar/list_calendar_events.py:79]: Sir checked and added s3 helper wherever required

File: superagi/helper/google_calendar_creds.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            expire_time = datetime.strptime(creds["expiry"], "%Y-%m-%dT%H:%M:%S.%fZ")
+            google_creds = session.query(ToolConfig).filter(ToolConfig.tool_kit_id == toolkit_id).all()
+            client_id = ""
+            client_secret = ""
+            for credentials in google_creds:
+                credentials = credentials.__dict__
+                if credentials["key"] == "GOOGLE_CLIENT_ID":
+                    client_id = credentials["value"]
+                else:
+                    client_secret = credentials["value"]
```
Comment (TransformerOptimus) [inline superagi/helper/google_calendar_creds.py:43]: you should make it more deterministic. Search exact key for secret and assign it

File: superagi/tools/google_calendar/list_calendar_events.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        file = file.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        file_name = "Google_Calendar_" + file + ".csv"
+        final_path = self.get_output_path(file_name)
+        with open(final_path, "w") as file:
+            writer = csv.writer(file, lineterminator="\n")
+            for row in csv_data:
+                writer.writerow(row)
+        return file_name
+
+    def get_output_path(self, file_name):
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/list_calendar_events.py:96]: use resource helper. New method created to fetch this.

**Review by TransformerOptimus — APPROVED** [review]

**Pull request merged**